client/src/pages/Students/ViewStudent.js

```diff
@@ -104,17 +107,36 @@ const ViewStudent = ({ student, editStudent, deleteStudent }) => {
                 </Flex>
 
                 <Flex p={4} flexDirection="column" marginTop="50px">
-                    <Flex flexDirection="row" align="center" justify="center">
-                        <Text textAlign="center" fontSize="2xl">
-                            Plans
-                        </Text>
-                        <Link to={`plan/${upi}/DefacedC0ffeeBabeB0b0001`}>
-                            <Button variantColor="blue" backgroundColor={c.uoaBlue} marginLeft="20px">
-                                <Text textAlign="center" color={c.white}>
-                                    Create Plan
-                                </Text>
-                            </Button>
-                        </Link>
+                    <Text textAlign="center" fontSize="2xl">
+                        Plans
+                    </Text>
+                    <Flex flexDirection="row" align="center" marginTop="10px" justify="center">
+                        <Select value={programmeID} ml={5} onChange={(e) => setProgrammeID(e.target.value)}>
+                            <option value={""}>Select a program..</option>
+                            {programmes.map((programme) => (
+                                <option value={programme._id}>{programme.name}</option>
+                            ))}
+                        </Select>
+                        <Button
+                            variantColor="blue"
+                            backgroundColor={c.uoaBlue}
+                            marginLeft="20px"
+                            isDisabled={!programmeID}
+                            onClick={() =>
+                                CoursePlannerClient.createStudentPlan(editUpi, {
+                                    name: `${name}'s ${programmes.find((programme) => programme._id === programmeID).name} plan`,
+                                    student: student._id,
+                                    programmeDegree: programmeID,
+                                    startYear: new Date().getFullYear() + 1,
+                                    endYear: new Date().getFullYear() + 1,
```

Comment:
Should Nasser be able to define these 0.o ?

Reply:
Instead of being hardcoded

Reply:
Is that covered by the edit plan

Reply:
No it is not at the moment

Reply:
I feel like it should default to current year and changes can be made on the plan page. Similar to how you create a new document on drive but edit the title afterwards.

Reply:
Okay that sounds good